sounddevice.py

```diff
@@ -2262,13 +2262,68 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : str, optional
+        User error message.
+    code : int, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
+
+    Notes
+    -----
+    - The `host_api`, `host_code`, and `host_text` attributes will be None
+      except for the case when `code` == `paUnanticipatedHostError`.
+    
+    Attributes
+    ----------
+    code : int or None
+        The PaErrorCode integer value or None if there is no associated error
+        code.
+    host_api : int or None
+        The index of the host API which returned the error code.
+    host_code : int or None
+        The host error code returned.
+    host_text : str or None
+        A textual description of the host error if available, otherwise a
+        zero-length string.
```

Comment:
I failed to mention this above, but I think if you combine something, it should be the three things, hostapi/hosterror_code/hosterror_text, because they belong together.
One without the other two doesn't make sense, right?